### ISSUE TITLE
FormsPlot: default right-click menu throws exception for some plot types

### ISSUE THREAD

**Issue by ShenxuanLi** [ScottPlot/ScottPlot#1791]
### System Details
**ScottPlot Version**:  4.1.41
**Operating System**:  Windows 10
**Application Type**:  WinForms
**NET Version**:  NET Framework 4.8

**no Crosshair is ok**
![3](https://user-images.githubusercontent.com/92280805/164397881-a616f52f-e8a5-4ca6-a166-4f0408acb0d5.png)

**if add a Crosshair, will crash**
![1](https://user-images.githubusercontent.com/92280805/164398366-e6057139-bfe2-4256-ab98-84fd3bfbd713.png)
![2](https://user-images.githubusercontent.com/92280805/164398183-6ada9b47-4219-4989-9ae5-fbc42ab246bc.png)

**Comment by ChunChunLiGong** [comment]
add a line or a span is ok. but add a crosshair will crash.

**Comment by ChunChunLiGong** [comment]
Thr version 4.1.30 is all right，but 4.1.41 with this bug.

**Comment by levimann24** [comment]
ChunChunLiGong I can confirm that am getting the same crash.

**Comment by levimann24** [comment]
@ChunChunLiGong I can confirm I am seeing the same bug.

**Comment by MareMare** [comment]
I think this is because the `Crosshair.GetLegendItems()` method returns null, but that is not included in the condition.

https://github.com/ScottPlot/ScottPlot/blob/f012da1a9ecb7effc4c83ab2cdcd7a049d0ba675/src/ScottPlot4/ScottPlot/Plottable/Crosshair.cs#L135

the current code:
https://github.com/ScottPlot/ScottPlot/blob/f012da1a9ecb7effc4c83ab2cdcd7a049d0ba675/src/ScottPlot4/ScottPlot.WinForms/FormsPlot.cs#L211-L214

I propose the above code as follows:
```cs
bool legendHasItems = Plot.GetPlottables()
    .Where(x => x.GetLegendItems() != null)  // <--- Add this as a condition
    .SelectMany(x => x.GetLegendItems())
    .Where(x => !string.IsNullOrEmpty(x.label))
    .Any();
```

What do you think?

**Comment by levimann24** [comment]
@MareMare Seems like a good solution!

**Comment by ChunChunLiGong** [comment]
@MareMare good solution！

**Comment by BigDream324** [comment]
I have the same problem and tried version 4.1.39 works.

**Comment by swharden** [comment]
Thanks for reporting this @ShenxuanLi! It has been fixed in #1794 and will be in the next version of ScottPlot released on NuGet shortly 👍
